test/junit/scala/collection/mutable/ArrayDequeTest.scala

```diff
@@ -104,18 +104,17 @@ class ArrayDequeTest {
 object ArrayDequeTest {
 
   // tests scala/bug#11047
-  def genericSlidingTest(factory: SeqFactory[ArrayDeque], collectionName: String): Unit =
+  def genericSlidingTest(factory: SeqFactory[ArrayDeque], collectionName: String): Unit = {
     for {
-      i <- 1 to 40
+      i <- 0 to 40
 
       range = 0 until i
-      iterable = collection.Iterable.from(range)
       other = factory.from(range)
 
       j <- 1 to 40
       k <- 1 to 40
 
-      iterableSliding = iterable.sliding(j,k).to(Seq)
+      iterableSliding = range.sliding(j, k).to(Seq)
       otherSliding = other.sliding(j, k).to(Seq)
     }
       assert(iterableSliding == otherSliding,
```

Comment:
Is the use of predef assert to avoid the eager message? Otherwise, `assertEquals` is more usual. Also FSR the assert is indented an extra tab.

Reply:
`assertEquals` doesn't allow customized error message, and I made quite a verbose one here to help debugging issues.

Reply:
From my terminal history:
```
[info] Test scala.collection.mutable.ArrayDequeTest.sliding started
[error] Test scala.collection.mutable.ArrayDequeTest.sliding failed: java.lang.AssertionError: Iterable.from(empty Range 0 until 0)).sliding(1,1) differs from ArrayDeque.from(empty Range 0 until 0)).sliding(1,1)
[error] Iterable yielded: ArrayBuffer()
[error] ArrayDeque yielded: ArrayBuffer(ArrayDeque())
[error]       expected:<ArrayBuffer(ArrayDeque())> but was:<ArrayBuffer()>, took 0.128 sec
```
It looks like I switched expected and actual by mistake. I'm not sure if you meant assertEquals doesn't take a string, or that it only takes an eager string, which is not desirable in general. Not suggesting to undertake it here. I won't insist on fixing indent, since you were kind enough to take my code style suggestion.